File: swarm-yuan/tests/fixtures/typeorm/compliant/src/service/user.service.ts
// compliant fixture 服务：
//  - where 用 Object 条件/参数绑定（防注入）
//  - 分页走 findAndCount take/skip（非 offset/limit）
//  - 软删除实体走 softDelete
import { AppDataSource } from '../data-source';
import { User } from '../entity/User';

export class UserService {
  async findByName(name: string, page = 1, pageSize = 20) {
    return AppDataSource.getRepository(User)
      .findAndCount({
        where: { name },
        relations: ['posts'],
        take: pageSize,
        skip: (page - 1) * pageSize,
      });
  }

  async removeUser(id: number) {
    await AppDataSource.getRepository(User).softDelete(id);
  }
}

File: swarm-yuan/tests/cross-stack-fixtures/monorepo/javaweb/backend/src/main/java/com/demo/x/UserService.java
package com.demo.x;

import com.demo.x.UserMapper;
import com.demo.x.domain.User;

public class UserService {
    private final UserMapper userMapper;
    public UserService(UserMapper userMapper) { this.userMapper = userMapper; }
    public User get(Long id) { return null; }
}
